### DISCUSSION TITLE
Adding the classpath to the MANIFEST.MF in the generated JAR

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-tycho/tycho#884] [Q&A]
I would like to add the classpath entries to the generated JAR file build via Tycho. Maven Archive allows this, see

https://maven.apache.org/shared/maven-archiver/examples/classpath.html

Is this supported in a Tycho build? I have a plug-in which also needs to be used outside OSGi.

Example: https://github.com/vogellacompany/tycho-example/blob/maven-jar-plugin/com.vogella.tycho.fatjar/pom.xml

Currently the https://github.com/vogellacompany/tycho-example/blob/maven-jar-plugin/com.vogella.tycho.fatjar/pom.xml#L45 entries does not make a difference.

**Comment by laeubi** [comment]
In your example it is unclear why do you need Tycho packaging at all, anyways you can bind to the `prepare-package` phase and extract whatever is required before packing it into the jar.

If you aim to modify the Archiver configuration of the Tycho packing you need to configure the tycho-packaging plugin:

https://www.eclipse.org/tycho/sitedocs/tycho-packaging-plugin/package-plugin-mojo.html#archive

**Reply by vogella** [reply]
> If you aim to modify the Archiver configuration of the Tycho packing you need to configure the tycho-packaging plugin:
> 
> https://www.eclipse.org/tycho/sitedocs/tycho-packaging-plugin/package-plugin-mojo.html#archive

Thanks. I see that <mainclass> configuration works if I configure it but not not the classpath entry. I used the configuration as described here: https://maven.apache.org/shared/maven-archiver/

See for the updated example: https://github.com/vogellacompany/tycho-example/blob/master/com.vogella.tycho.fatjar/pom.xml#L50

Is this a missing feature in the Tycho / Maven Archiver integration?

**Comment by laeubi** [comment]
> I have a plug-in which also needs to be used outside OSGi.

Just a note: Only because something is used outside OSGi does not mean you need to package everything into one fat jar...

**Comment by vogella** [comment]
I created https://github.com/eclipse/tycho/issues/997 for this missing feature / bug

**Reply by wallenda** [reply]
it solved？？ @vogella
